open-api/rest-catalog-open-api.yaml

```diff
@@ -2106,6 +2204,28 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        An opaque JSON object that contains information provided by the REST server,
+        to be utilized by clients for distributed table scan planning, should be supplied
+        as is for input in `PlanTable` operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        delete-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/ContentFile'
+        residual-filter:
+          description: filters that are not fully applied by the server and should be pushed down to the file reader when reading the file.
```

Comment:
I left a comment about how to interpret this when the residual is missing. I think that needs to be here to document expected behavior.

Reply:
Will add that line back.